libr/anal/p/anal_evm_cs.c

```diff
@@ -32,8 +39,59 @@ static void set_opdir(RAnalOp *op) {
 	}
 }
 
+/* Jumps/calls in EVM are done via first pushing dst value
+ * on the stack, and then calling a jump/jumpi instruction, for example:
+ *   0x0000000d push 0x42
+ *   0x0000000f jumpi
+ *
+ * we are storing the value in push instruction to db, but not at the
+ * addr of the push instruction, but at the addr of next jumpi instruction.
+ * So in our example we are inserting (0xf, 0x42)
+ */
+static int evm_add_push_to_db(RAnalOp *op, ut64 addr, const ut8 *buf, int len) {
+	ut64 next_cmd_addr = 0;
+	ut64 dst_addr = 0;
+	size_t i, push_size;
+	char key[16] = {0}, value[16] = {0};
+
+	push_size = op->id - EVM_INS_PUSH1;
+	next_cmd_addr = addr + push_size + 2;
+
+	for (i = 0; i < push_size + 1; i++) {
+		dst_addr <<= 8;
+		dst_addr |= buf[i + 1];
+	}
+
+	if (evm_ai) {
+		snprintf (key, sizeof(key) - 1, "%08x", (unsigned) next_cmd_addr);
+		snprintf (value, sizeof(value) - 1, "%08x", (unsigned) dst_addr);
+		sdb_set (evm_ai->pushs_db, key, value, 0);
+	}
+
+	return 0;
+}
+
+static st64 evm_get_jmp_addr(ut64 addr) {
+	char key[16] = {0};
+	const char *value;
+	unsigned ret;
+
+	snprintf (key, sizeof(key) - 1, "%08x", (unsigned) addr);
+
+	value = sdb_const_get (evm_ai->pushs_db, key, 0);
```

Comment:
i have implemented ut64 as key apis in sdb so now you can do `ut64 value = sdb_nget (evm_ai->pushs_db, addr);`